components/renku_data_services/data_connectors/db.py

```diff
@@ -297,6 +300,160 @@ async def delete_data_connector(
         return data_connector
 
 
+class DataConnectorProjectLinkRepository:
+    """Repository for links from data connectors to projects."""
+
+    def __init__(
+        self,
+        session_maker: Callable[..., AsyncSession],
+        authz: Authz,
+    ) -> None:
+        self.session_maker = session_maker
+        self.authz = authz
+
+    async def get_links_from(
+        self, user: base_models.APIUser, data_connector_id: ULID
+    ) -> list[models.DataConnectorToProjectLink]:
+        """Get links from a given data connector."""
+        authorized = await self.authz.has_permission(user, ResourceType.data_connector, data_connector_id, Scope.READ)
+        if not authorized:
+            raise errors.MissingResourceError(
+                message=f"Data connector with id '{
+                    data_connector_id}' does not exist or you do not have access to it."
+            )
+
+        project_ids = await self.authz.resources_with_permission(user, user.id, ResourceType.project, Scope.READ)
+
+        async with self.session_maker() as session:
+            stmt = (
+                select(schemas.DataConnectorToProjectLinkORM)
+                .where(schemas.DataConnectorToProjectLinkORM.data_connector_id == data_connector_id)
+                .where(schemas.DataConnectorToProjectLinkORM.project_id.in_(project_ids))
+            )
+            result = await session.scalars(stmt)
+            links_orm = result.all()
+            return [link.dump() for link in links_orm]
+
+    async def get_links_to(
+        self, user: base_models.APIUser, project_id: ULID
+    ) -> list[models.DataConnectorToProjectLink]:
+        """Get links to a given project."""
+        authorized = await self.authz.has_permission(user, ResourceType.project, project_id, Scope.READ)
+        if not authorized:
+            raise errors.MissingResourceError(
+                message=f"Project with id '{
+                    project_id}' does not exist or you do not have access to it."
+            )
```

Comment:
Wrong format.

Reply:
Done in 968df0f89befff1c2e5564770fdccd0638479432.